sql/core/src/test/resources/sql-tests/results/cte-ddl.sql.out

```diff
@@ -0,0 +1,1152 @@
+Schema: root
+ |-- a: integer (nullable = true)
+ |-- b: string (nullable = true)
+ |-- c: integer (nullable = true)
+
+
+-- !query
+WITH s AS (SHOW TABLE EXTENDED LIKE 'test_show_table*') SELECT isTemporary FROM s WHERE tableName = 'test_show_tables'
+-- !query schema
+struct<isTemporary:boolean>
+-- !query output
+false
+
+
+-- !query
+WITH s(ns, tn, t, info) AS (SHOW TABLE EXTENDED LIKE 'test_show_table*') SELECT * FROM s
+-- !query schema
+struct<ns:string,tn:string,t:boolean,info:string>
+-- !query output
+query_ddl_namespace	test_show_table_properties	false	Database: query_ddl_namespace
+Table: test_show_table_properties
+Created Time [not included in comparison]
+Last Access [not included in comparison]
+Created By [not included in comparison]
+Type: MANAGED
+Provider: parquet
+Table Properties: [p1=v1, p2=v2]
+Location [not included in comparison]/{warehouse_dir}/query_ddl_namespace.db/test_show_table_properties
+Schema: root
+ |-- a: integer (nullable = true)
+ |-- b: string (nullable = true)
+ |-- c: integer (nullable = true)
+
+
+query_ddl_namespace	test_show_tables	false	Database: query_ddl_namespace
+Table: test_show_tables
+Created Time [not included in comparison]
+Last Access [not included in comparison]
+Created By [not included in comparison]
+Type: MANAGED
+Provider: parquet
+Location [not included in comparison]/{warehouse_dir}/query_ddl_namespace.db/test_show_tables
+Schema: root
+ |-- a: integer (nullable = true)
+ |-- b: string (nullable = true)
+ |-- c: integer (nullable = true)
+
+
+-- !query
+WITH s(ns, tn, t, info) AS (SHOW TABLE EXTENDED LIKE 'test_show_table*') SELECT t FROM s
+-- !query schema
+struct<t:boolean>
+-- !query output
+false
+false
+
+
+-- !query
+WITH s(ns, tn, t, info) AS (SHOW TABLE EXTENDED LIKE 'test_show_table*') SELECT * FROM s WHERE tn = 'test_show_tables'
+-- !query schema
+struct<ns:string,tn:string,t:boolean,info:string>
+-- !query output
+query_ddl_namespace	test_show_tables	false	Database: query_ddl_namespace
+Table: test_show_tables
+Created Time [not included in comparison]
+Last Access [not included in comparison]
+Created By [not included in comparison]
+Type: MANAGED
+Provider: parquet
+Location [not included in comparison]/{warehouse_dir}/query_ddl_namespace.db/test_show_tables
+Schema: root
+ |-- a: integer (nullable = true)
+ |-- b: string (nullable = true)
+ |-- c: integer (nullable = true)
+
+
+-- !query
+WITH s(ns, tn, t, info) AS (SHOW TABLE EXTENDED LIKE 'test_show_table*') SELECT t FROM s WHERE tn = 'test_show_tables'
+-- !query schema
+struct<t:boolean>
+-- !query output
+false
+
+
+-- !query
+WITH s AS (SHOW TABLE EXTENDED LIKE 'test_show_table*') SELECT * FROM s ORDER BY tableName
+-- !query schema
+struct<>
+-- !query output
+java.lang.ClassCastException
+org.apache.spark.sql.catalyst.expressions.GenericInternalRow cannot be cast to org.apache.spark.sql.catalyst.expressions.UnsafeRow
```

Comment:
I don't know the reason yet. It seems we should fix this issue in another PR?
cc @cloud-fan @wangyum @maropu @viirya